wetmap/src/components/newModals/shopModal/index.tsx

```diff
@@ -1,105 +1,102 @@
-  // const {lat, lng, setSelectedPhoto, setPhotoBoxModel } = props
-  // const { shopModal, setShopModal } = useContext(ShopModalContext);
-  // const { selectedShop, setSelectedShop } = useContext(SelectedShopContext);
-  // const [siteCloseState, setSiteCloseState] = useState(false);
-  // const [itineraryList, setItineraryList] = useState('');
-  // const [selectedID, setSelectedID] = useState(null);
-  // const { masterSwitch, setMasterSwitch } = useContext(MasterContext);
-  // const { mapCoords, setMapCoords } = useContext(CoordsContext);
-  // const { zoomHelper, setZoomHelper } = useContext(ZoomHelperContext);
-
-  // useEffect(() => {
-  //   if (selectedShop[0]) {
-  //     getItineraries(selectedShop[0].id);
-  //     setMasterSwitch(true);
-  //   }
-  // }, [selectedShop]);
-
-  // useEffect(() => {
-  //   if (shopModal && zoomHelper) {
-  //     setMapCoords([selectedShop[0].lat, selectedShop[0].lng]);
-  //   }
-  // }, [shopModal]);
-
-  // const getItineraries = async (IdNum) => {
-  //   try {
-  //     const itins = await itineraries(IdNum);
-  //     if (itins.length > 0) {
-  //       setItineraryList(itins);
-  //     }
-  //   }
-  //   catch (e) {
-  //     console.log({ title: 'Error', message: e.message });
-  //   }
-  // };
-
-  // const handleShopModalClose = () => {
-  //   setSelectedShop({ ...selectedShop, id: 0, orgName: '' });
-  //   setItineraryList('');
-  //   setShopModal(false);
-  // };
-
   const { shopModal, setShopModal } = useContext(ShopModalContext);
   const { selectedShop, setSelectedShop } = useContext(SelectedShopContext);
-  const [siteCloseState, setSiteCloseState] = useState(false);
-  const [itineraryList, setItineraryList] = useState('');
-  const [selectedID, setSelectedID] = useState(null);
+  const { profile } = useContext(UserProfileContext);
   const { masterSwitch, setMasterSwitch } = useContext(MasterContext);
   const { mapCoords, setMapCoords } = useContext(CoordsContext);
   const { zoomHelper, setZoomHelper } = useContext(ZoomHelperContext);
 
+  const [isPartnerAccount, setIsPartnerAccount] = useState(false);
+  const [itineraryList, setItineraryList] = useState<ItineraryItem[]>([]);
+  const [selectedID, setSelectedID] = useState<number>(0);
+
   useEffect(() => {
     if (selectedShop[0]) {
       getItineraries(selectedShop[0].id);
       setMasterSwitch(true);
     }
   }, [selectedShop]);
 
+  useEffect(() => {
+    if (profile[0].partnerAccount) {
+      setIsPartnerAccount(true);
+    }
+  }, []);
+
   useEffect(() => {
     if (shopModal && zoomHelper) {
       setMapCoords([selectedShop[0].lat, selectedShop[0].lng]);
     }
   }, [shopModal]);
 
-  const getItineraries = async (IdNum) => {
+  const getItineraries = async (IdNum: number) => {
     try {
       const itins = await itineraries(IdNum);
-      if (itins.length > 0) {
-        setItineraryList(itins);
+      if (itins && itins.length > 0) {
+        setItineraryList(itins.map(itin => ({
```

Comment:
It's not a big deal. I'm just trying to understand why you've copied each property. If it's just for readability and make it clear what's in that state maybe it's better to use types: convert itinerarySupabaseCalls to ts and use "ItineraryItem" type that you created as a type of return in itinerarySupabaseCalls.itineraries.
Let me know what you think.

Reply:
That sounds like a great idea. I did it specifically to ensure that the type mapped to each itinerary item was of type ItineraryItem, defined in types. It was complaining that each of the itins was of the wrong type.

I'll try to get your suggestion working as it is more clean

![image](https://github.com/user-attachments/assets/1a67582a-2f7b-458f-a50c-da95e15b769f)

Reply:
Thanks for the suggestion. Made the changes